router.go

```diff
@@ -375,8 +379,8 @@ func resolvePathToFieldDescriptors(
 		if field == nil {
 			field = fields.ByName(protoreflect.Name(part))
 			if field == nil {
-				return nil, fmt.Errorf("in field path %q: element %q does not correspond to any field of type %s",
-					path, part, msg.FullName())
+				return nil, fmt.Errorf("in field path %q: element %q does not correspond to any field of type %s: %w",
+					path, part, msg.FullName(), errUnknownField)
```

Comment:
I think moving the order of the wrapped error will produce a better final error message.
```suggestion
				return nil, fmt.Errorf("%w in field path %q: element %q does not correspond to any field of type %s",
					errUnknownField, path, part, msg.FullName())
```